src/leo_nlm_proto4_server.erl

```diff
@@ -0,0 +1,314 @@
+%%======================================================================
+%%
+%% Leo Gateway
+%%
+%% Copyright (c) 2012-2015 Rakuten, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%======================================================================
+-module(leo_nlm_proto4_server).
+
+-include("leo_nlm_proto4.hrl").
+-include_lib("leo_logger/include/leo_logger.hrl").
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+-export([nlmproc4_null_4/2,
+         nlmproc4_test_4/3,
+         nlmproc4_lock_4/3,
+         nlmproc4_cancel_4/3,
+         nlmproc4_unlock_4/3
+        ]).
+
+%% nlmproc4_granted_4/3,
+%% nlmproc4_test_msg_4/3,
+%% nlmproc4_lock_msg_4/3,
+%% nlmproc4_cancel_msg_4/3,
+%% nlmproc4_unlock_msg_4/3,
+%% nlmproc4_granted_msg_4/3,
+%% nlmproc4_test_res_4/3,
+%% nlmproc4_lock_res_4/3,
+%% nlmproc4_cancel_res_4/3,
+%% nlmproc4_unlock_res_4/3,
+%% nlmproc4_granted_res_4/3,
+%% nlmproc4_share_4/3,
+%% nlmproc4_unshare_4/3,
+%% nlmproc4_nm_lock_4/3,
+%% nlmproc4_free_all_4/3
+
+-record(lock_record,{
+          start :: non_neg_integer(),
+          till  :: integer(),
```

Comment:
As `end` is a reserved word in Erlang, `till` does not look good to me either but I simply out of idea for alternatives....

`-1` is used to represent locking range until EOF so integer() is used